vim-patch:8.0.1179: Test_popup_and_window_resize() does not always pass

blueyed opened the pull request:
Problem:    Test_popup_and_window_resize() does not always pass.
Solution:   Do not use $VIMPROG, pass the Vim executable in the vimcmd file.
            (Ozaki Kiichi, closes vim/vim#2186)
https://github.com/vim/vim/commit/631820536e4084e01bf990f9314ec385b60b21d7

NOTE: Does not add `GetVimProg()`, but keeps/syncs existing usages of it
in tests (which obviously are all skipped, mostly because of
`!has('terminal')`).  Likely to be replaced with `v:progpath` in Vim [1], or
at least that would/should be used with Neovim then probably.

1: https://github.com/vim/vim/pull/4806

blueyed committed:
vim-patch:8.0.1179: Test_popup_and_window_resize() does not always pass

Problem:    Test_popup_and_window_resize() does not always pass.
Solution:   Do not use $VIMPROG, pass the Vim executable in the vimcmd file.
            (Ozaki Kiichi, closes vim/vim#2186)
https://github.com/vim/vim/commit/631820536e4084e01bf990f9314ec385b60b21d7

vim-patch:8.0.1526: no test using a screen dump yet

Problem:    No test using a screen dump yet.
Solution:   Add a test for C syntax highlighting.  Add helper functions.
https://github.com/vim/vim/commit/da65058a9c4774dc534c7ae98d24c58b5db669fa

NOTE: uses modified `GetVimProg()` (which is used with skipped tests only
(mostly because of `!has('terminal')`)).

Vim uses a 'vimcmd' file, while Nvim uses `$NVIM_TEST_ARGX` environment
variables.

Ref: https://github.com/vim/vim/pull/4806

janlazo commented:
Add https://github.com/vim/vim/commit/9570aacdb82c1eb5645d0fbc35bf9728d7334b23 here.

blueyed committed:
vim-patch:8.1.1842: test listed as flaky should no longer be flaky

Problem:    Test listed as flaky should no longer be flaky.
Solution:   Remove Test_popup_and_window_resize from the list of flaky tests.
            (Daniel Hahler, close vim/vim#4807)
https://github.com/vim/vim/commit/9570aacdb82c1eb5645d0fbc35bf9728d7334b23

blueyed commented:
Added "patch 8.0.1526: no test using a screen dump yet" also.  Amended `GetVimProg()`.

blueyed commented:
Note: unlike on Vim (https://github.com/vim/vim/pull/4806#issuecomment-520623678), `v:progpath` is absolute always (https://github.com/blueyed/neovim/blob/8849011fbe10c8f348a54fbc6167a7625a7ce733/test/functional/options/defaults_spec.lua#L195-L198).

justinmk commented on src/nvim/testdir/shared.vim:
```diff
@@ -196,16 +196,26 @@ func s:feedkeys(timer)
+func GetVimProg()
+  if empty($NVIM_TEST_ARG0)
+    " Assume the script was sourced instead of running "make".
+    return '../../../build/bin/nvim'
```
could use v:progpath then

blueyed replied:
This tries to follow the (not really clear) use case mentioned in https://github.com/vim/vim/pull/4806#issuecomment-520847278.
